### ISSUE TITLE
paramType does not contain attribute element in compound.xsd schema  (Origin: bugzilla #304793)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#1634]
status NEW severity *minor* in component *build* for *---*
Reported in version *1.4.x* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[compound.xsd](https://sourceforge.net/projects/doxygen/files/attachments/bug304793_46648_compound.xsd/download/download)_ (ID 46648)
- _[interface_i_application.xml](https://sourceforge.net/projects/doxygen/files/attachments/bug304793_46649_interface_i_application.xml/download/download)_ (ID 46649)

On 2005-05-19 18:30:41 +0000, Michael S. Scherotter wrote:

    Using a validating parser (like MSXML4), the XML generated by Doxygen is 
    invalid for param elements in compounds:
    In an xml file that has a compounddef type, a <param> element has an 
    <attributes> element as its first child.
    In the file compound.xsd the definition of complexType "paramType" has
    "type"
    as the first element in its sequence of child elements, with no "attributes"
    in
    the sequence at all.

On 2005-05-19 18:31:53 +0000, Michael S. Scherotter wrote:

    Created attachment 46648
    Compound schema

On 2005-05-19 18:33:28 +0000, Michael S. Scherotter wrote:

    Created attachment 46649
    XML file generated by doxygen
    
    Error message when parsing with MSXML4:
    Error -2147467259: Error while parsing "file:///C:/Projects/SMP
    Solutions/MM6
    Object Model/Output/xml/interface_i_application.xml". Element content is
    invalid according to the DTD/Schema.
    Expecting: type, declname, defname, array, defval, briefdescription.

**Comment by kevinlatimer** [comment]
Here are some fixes I've found so far that are needed to adapt to modern Objective-C:

In `compound.xsd`:

- `<xsd:element name="attributes" minOccurs="0" />` needs to be added as the first sequence element to `<xsd:complexType name="paramType">...</xsd:complexType>`

- These need to be added to `<xsd:complexType name="memberdefType">`:
`<xsd:attribute name="privategettable" type="DoxBool" use="optional"/>`
`<xsd:attribute name="protectedgettable" type="DoxBool" use="optional"/>`
`<xsd:attribute name="privatesettable" type="DoxBool" use="optional"/>`
`<xsd:attribute name="protectedsettable" type="DoxBool" use="optional"/>`

**Comment by albert-github** [comment]
@kevinlatimer 
Thank you for the information .
- which version of doxygen did you use
- in the current master it looks to me that  getable / settable items are already included (see merged pull request: #6890)
- regarding the `<xsd:element name="attributes" minOccurs="0" />` could you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7035

**Comment by kevinlatimer** [comment]
The patches to the schemas I had added were made with respect to version 1.8.15 which was what I had downloaded yesterday from the Doxygen downloads page. Unfortunately I cannot upload any source as it contains work product, but specifically with regards to `<xsd:element name="attributes" minOccurs="0" />`, I believe you will encounter this by parsing something similar to:
```
@protocol myProtocol
- (void)myVirtualMethod;
@end

@interface myClass : NSObject
- (void)myMethod:(id<myProtocol>) myParameter;
@end
```
When parsed, attributes are mapped to the protocol(s) (in this case `myProtocol`) which constrain the type (in this case, `id`)

**Comment by albert-github** [comment]
I didn't try your example yet (thanks!), but as you can see I already made a proposed patch to have the "attributes" included (based on your message, the (output) files of the original issue and reading the code).

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
